sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/IEDAdapter.java

```diff
@@ -375,10 +381,40 @@ public Optional<TCompasICDHeader> getCompasICDHeader() {
 
     /**
      * Get value of private type COMPAS-SystemVersion
+     *
      * @return COMPAS-SystemVersion private value if present, else empty Optional
      */
     public Optional<TCompasSystemVersion> getCompasSystemVersion() {
         return PrivateService.extractCompasPrivate(currentElem, TCompasSystemVersion.class);
     }
 
+    /**
+     * Update and/or create Monitoring LNs (LSVS and LGOS) into LDSUIED for each bound ExtRef of each LDevice
+     *
+     * @return a list of SclReport Objects that contains errors
+     */
+    public List<SclReportItem> manageMonitoringLns() {
+        List<SclReportItem> sclReportItems = new ArrayList<>();
+        findLDeviceAdapterByLdInst(LDSUIED_LDINST).ifPresent(lDeviceAdapter -> {
+            lDeviceAdapter.manageMonitoringLns(retrieveAllExtRefForServiceType(TServiceType.GOOSE), DO_GOCBREF, MonitoringLnClassEnum.LGOS)
+                    .ifPresent(sclReportItems::add);
+            lDeviceAdapter.manageMonitoringLns(retrieveAllExtRefForServiceType(TServiceType.SMV), DO_SVCBREF, MonitoringLnClassEnum.LSVS)
+                    .ifPresent(sclReportItems::add);
+        });
+        return sclReportItems;
+    }
+
+    private List<TExtRef> retrieveAllExtRefForServiceType(TServiceType tServiceType) {
+        List<TExtRef> tGseExtRefs = streamLDeviceAdapters()
+                .map(LDeviceAdapter::getLN0Adapter)
+                .map(LN0Adapter::getExtRefs)
+                .flatMap(List::stream)
+                .filter(tExtRef -> tExtRef.isSetServiceType() && tExtRef.isSetSrcCBName() &&
+                        tServiceType.equals(tExtRef.getServiceType()))
+                .toList();
+
+        return filterDuplicatedExtRefs(tGseExtRefs);
```

Comment:
As a result if we filter duplicated ExtRefs is InputsAdapter.
```suggestion
        return streamLDeviceAdapters()
                .map(LDeviceAdapter::getLN0Adapter)
                .map(LN0Adapter::getInputsAdapter)
                .map(InputsAdapter::filterDuplicatedExtRefs) // filter duplicated ExtRefs is InputsAdapter.
                .flatMap(List::stream)
                .filter(tExtRef -> tExtRef.isSetServiceType() && tExtRef.isSetSrcCBName() &&
                        tServiceType.equals(tExtRef.getServiceType()))
                .toList();
    }
```